FileLiberator/DownloadBook.cs

```diff
@@ -40,7 +41,53 @@ private static string getDownloadPath(LibraryBook libraryBook)
 				libraryBook.Book.Title,
 				"aax",
 				libraryBook.Book.AudibleProductId);
+		private async Task<string> downloadAacxBookAsync(LibraryBook libraryBook, string tempAaxFilename)
+		{
+			validate(libraryBook);
+
+			var api = await GetApiAsync(libraryBook);
+
+			var dlLic = await api.GetDownloadLicenseAsync(libraryBook.Book.AudibleProductId);
+
+			libraryBook.Book.AudibleKey = dlLic.AudibleKey;
+			libraryBook.Book.AudibleIV = dlLic.AudibleIV;
+
+			var client = new HttpClient();
+			client.DefaultRequestHeaders.Add("User-Agent", Resources.UserAgent);
+
+			var actualFilePath = await PerformDownloadAsync(
+				tempAaxFilename,
+				(p) => client.DownloadFileAsync(dlLic.DownloadUri.AbsoluteUri, tempAaxFilename, p));
 
+			System.Threading.Thread.Sleep(100);
+			// if bad file download, a 0-33 byte file will be created
+			// if service unavailable, a 52 byte string will be saved as file
+			var length = new FileInfo(actualFilePath).Length;
+
+			if (length > 100)
+				return actualFilePath;
+
+			var contents = File.ReadAllText(actualFilePath);
+			File.Delete(actualFilePath);
+
+			var exMsg = contents.StartsWithInsensitive(SERVICE_UNAVAILABLE)
+				? SERVICE_UNAVAILABLE
+				: "Error downloading file";
+
+			var ex = new Exception(exMsg);
+			Serilog.Log.Logger.Error(ex, "Download error {@DebugInfo}", new
+			{
+				libraryBook.Book.Title,
+				libraryBook.Book.AudibleProductId,
+				libraryBook.Book.Locale,
+				Account = libraryBook.Account?.ToMask() ?? "[empty]",
+				tempAaxFilename,
+				actualFilePath,
+				length,
+				contents
+			});
+			throw ex;
+		}
 		private async Task<string> downloadBookAsync(LibraryBook libraryBook, string tempAaxFilename)
```

Comment:
If unused, this can be deleted

Reply:
Follow up to previous comments -- it looks like this should still be used for aax. The poorly named property "IsNonLibraryAudiblePlus" can help here